command/agent/config/config.go

```diff
@@ -1035,3 +1069,126 @@ func parseTemplates(result *Config, list *ast.ObjectList) error {
 	result.Templates = tcs
 	return nil
 }
+
+func parseExec(result *Config, list *ast.ObjectList) error {
+	name := "exec"
+
+	execList := list.Filter(name)
+	if len(execList.Items) == 0 {
+		return nil
+	}
+
+	if len(execList.Items) > 1 {
+		return fmt.Errorf("at most one %q block is allowed", name)
+	}
+
+	item := execList.Items[0]
+	var shadow interface{}
+	if err := hcl.DecodeObject(&shadow, item.Val); err != nil {
+		return fmt.Errorf("error decoding config: %s", err)
+	}
+
+	parsed, ok := shadow.(map[string]interface{})
+	if !ok {
+		return errors.New("error converting config")
+	}
+
+	var ec ExecConfig
```

Comment:
Nit: this name is a bit ambiguous since it's used far from declaration, maybe something like `exec` or `execConfig` would be better?

Reply:
its the same style as the other parse functions 🤷